graphql-spring-boot-test/src/main/java/com/graphql/spring/boot/test/RequestFactory.java

```diff
@@ -19,8 +19,8 @@ private RequestFactory() {
     static HttpEntity<Object> forMultipart(String query, String variables, HttpHeaders headers) {
         headers.setContentType(MediaType.MULTIPART_FORM_DATA);
         LinkedMultiValueMap<String, Object> values = new LinkedMultiValueMap<>();
-        values.add("query", forJson(query, headers));
-        values.add("variables", forJson(variables, headers));
+        values.add("query", forJson(query, new HttpHeaders()));
```

Comment:
This overrides any customization people would want to do with the headers, and shouldn't be necessary seeing my comment above. So would assume the cause of your particular problem is somewhere else.

Reply:
Mmh I assume people would want to customize the general headers to pass authentication for instance, which they still can. It's not that common to have the need to customize the sub-parts of the request IMO.